ios/engine/KMEI/KeymanEngine/Classes/kmp.json/KMPKeyboard.swift

```diff
@@ -41,6 +57,27 @@ class KMPKeyboard: Codable, KMPResource {
     languages = try values.decode([KMPLanguage].self, forKey: .languages)
   }
 
+  func matches(installable resource: LanguageResource, requireLanguageMatch: Bool = true) -> Bool {
+    guard let resource = resource as? InstallableKeyboard else {
+      return false
+    }
+
+    if id != resource.id {
+      return false
+    } else if version != resource.version {
+      return false
+    }
+
+    if requireLanguageMatch {
+      let resourceMetadata = KMPKeyboard(from: resource)!
+      return languages.contains(where: { language in
+        return language.languageId == resourceMetadata.languages[0].languageId
```

Comment:
Are you only checking the language ID for index 0? What if the matching language ID is later in the array?

Reply:
There is no "later in the array" in this case.  That's on a `KMPKeyboard` as generated from a single `InstallableResource`, which is guaranteed to only have a single language entry.

It's when a `KMPKeyboard` is loaded from `kmp.json` that it may have multiple entries.

Reply:
For example, for `nrc.en.mtnt`, there are three supported language codes:  `en`, `en-us`, and `en-ca`, I believe.  Yes, "lexical model" and not "keyboard", but the code is parallel.

`KMPKeyboard(from: resource)` will take in only a single `InstallableLexicalModel` (which in Java terms "implements" `LanguageResource`), which directly matches **one** of those three codes.

This method is generally called on `KMPKeyboard` instances decoded from a `kmp.json` file.  For this example, we've loaded `nrc.en.mtnt`'s `kmp.json`.  It has all three - so `languages.contains(where:` iterates over the language array found in the lexical model's entry in the _actual_ `kmp.json` - the array that actually has all three entries.  (`en`, `en-us`, and `en-ca`)